lib/internal/url.js

```diff
@@ -174,6 +178,20 @@ function isURLSearchParams(self) {
   return self && self[searchParams] && !self[searchParams][searchParams];
 }
 
+// WPT needs Error in URLSearchParams' constructor exactly be an instance of
+// TypeError.
+function throwTypeError(message) {
+  if (message instanceof Error) {
+    const err = new TypeError(message.message); // eslint-disable-line
+    err.stack = message.stack;
+    if (message.code) err.code = message.code;
+    if (message.name) err.name = message.name;
+    throw err;
+  }
+
+  throw new TypeError(message); // eslint-disable-line
+}
+
```

Comment:
This is not the correct way of handling this. If you take a look at `lib/internals/errors.js`, you'll see that some error codes have multiple types in their list... for instance `ERR_INVALID_STATE` lists `Error`, `TypeError`, and `RangeError`. To create an `ERR_INVALID_STATE` error this is a `TypeError`, you simply do `new ERR_INVALID_STATE.TypeError("message")`